### ISSUE TITLE
[Bug] `r` replacement in Word loses character's special marks

### ISSUE THREAD

**Issue by gilbh** [pit-ray/win-vind#110]
In Word, when I click `r` to replace a character, the new character will be inserted without special marks. For example, if I have a word in italics:
_hallo_
-- and I wish to replace the `a` with an `e` using the `r` command, the result will be:
_h_ e _llo_ (Note the e is not italicized; I added spaces around the `e`)

This also happens with diacritics: typing `r` command and then a character with a diacritical mark (e.g., `ā`) will result in a regular `a` (the overline/macron sign will not appear).

Thanks!

**Comment by pit-ray** [comment]
Hi.
In my environment, the `r` command insert a replaced character with the italic style.

Is the italic hello string generated by selecting the "hello" text and input CTRL+I shortcut key?

I show my environment recordings as follows.
![win-vind-r-replacing-demo](https://user-images.githubusercontent.com/42631247/193449765-c3f3260d-ad8f-4a42-9e08-8e6aae6c3628.gif)

Thanks.

**Comment by gilbh** [comment]
Indeed. I understand the problem better now, and it is not exactly as I described earlier. 

I have a Word template with shortcuts to diacritics, e.g. <ctrl-e> will insert `ē`. Now, after I press `r` to replace a letter, I hit <ctrl-e> and I get `e` instead of `ē`. I think the problem is that the <ctrl> key is ignored by win-vind after the `r` operation.

Is this correct and is there a way to fix this?

Thanks!

**Comment by pit-ray** [comment]
@gilbh 
In order to implement the looping of characters by prefix numbers, win-vind converts the input characters once into a unicode string. Then, the string is converted based on the key state at the time of the temporary input rather than sequential input.
https://github.com/pit-ray/win-vind/blob/e63e581ce50268dabeb3dbaa9e7589f3165d8143/src/bind/emu/replacetext.cpp#L81

I think it is possible to implement if they make a sequential decision.

Thanks.

**Comment by gilbh** [comment]
Thanks and no worries about reply time. Vind is becoming better and better all the time!

I am not sure I get the technical explanation you give here, but let me know if you want me to test a version.
